src/catchup/ApplyCheckpointWork.cpp

```diff
@@ -89,6 +92,7 @@ ApplyCheckpointWork::openInputFiles()
     mHdrIn.open(hi.localPath_nogz());
     mTxIn.open(ti.localPath_nogz());
     mTxHistoryEntry = TransactionHistoryEntry();
+    mHeaderHistoryEntry = LedgerHeaderHistoryEntry();
```

Comment:
I made it a member variable because the entry needs to be saved while `mConditionalWork` is running. An invariant is checked once the work finishes at the top of `applyHistoryOfSingleLedger`.